ivy/functional/frontends/torch/nn/functional/loss_functions.py

```diff
@@ -214,3 +214,17 @@ def nll_loss(
     ret = reduct(loss)
 
     return ret
+
+
+@to_ivy_arrays_and_back
+def soft_margin_loss(
```

Comment:
The tests will fail for `float16` we need to specify unsupported dtypes
```suggestion
@with_unsupported_dtypes({"1.11.0 and below": ("float16", "bfloat16")}, "torch")
def soft_margin_loss(
```